test/goose/brokers/redis/integration_test.clj

```diff
@@ -191,14 +191,17 @@
 (def batch-args-count (atom 0))
 (def batch-args-acc (atom (vector)))
 (def perform-batch-fn-executed (atom (promise)))
+(def callback-fn-executed (atom (promise)))
 (defn perform-batch-fn [arg]
   (swap! batch-args-acc conj (vector arg))
   (swap! batch-exec-count inc)
   (when (= @batch-exec-count @batch-args-count)
     (deliver @perform-batch-fn-executed @batch-args-acc)))
+(defn perform-batch-callback-fn [id m]
+  (deliver @callback-fn-executed (assoc m :id id)))
 
-; TODO: replace nil with a callback-fn-sym when callback is implemented
-(def batch-opts {:callback-fn-sym nil})
+(def batch-opts {:callback-fn-sym `perform-batch-callback-fn
```

Comment:
We should add another integration test for when a batch completes with dead jobs to test `post-batch-exec` being called from `catch` block of middleware